### PR TITLE
Export working copy

### LINKED ISSUE #14: Core: Export a working copy of a record

### PR DESCRIPTION
**Pull request by basilesimon**
Closes #14 

The UI should provide the option for a user to download a ZIP file of the record.

To-do:

- [ ] Write timestamp of export to ledger
- [x] Includes in ZIP a text file with metadata about the download

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -56,6 +59,39 @@ export const newBundle = (b: Bundle.newBundle): Promise<Bundle.Bundle> => {
...
+  return new Promise<void>((resolve, reject) => {
+    zip
+      .append(fs.createReadStream(`${root}/${screenshot}`), {
+        name: screenshot,
+      })
+      .append(fs.createReadStream(`${root}/${one_file}`), { name: one_file })
+      .on('error', err => reject(err))
+      .pipe(stream);
+
+    zip.finalize().then(() => {
```
Comment (nikonikoniko) [inline src/store/index.ts:88]: I am pretty sure the problem is here. according to these docs (well, the comment at the bottom of): https://github.com/archiverjs/node-archiver

`finalize` is not actually called when it is done, rather some other event with streams that I don't understand.

I believe what you want is something along the lines of
```
stream.on('close', () => resolve()
```

File: index.ts [hunk 2]
```diff
@@ -48,6 +48,30 @@ app.get(
...
+app.get(
+  '/export-copy/:sku',
+  async (req: Request, res: Response): Promise<void> => {
+    const { sku } = req.params;
+    const options = {
+      root: join(__dirname, './out'),
+      dotfiles: 'deny',
+    };
+    await Ledger.getDoc(sku)
+      .then((b: Record.Record) => Store.makeZip(b.bundle))
```
Comment (nikonikoniko) [inline index.ts:60]: the problem is deffinitely that `Store.makezip` is returning early, or not returning a promise at all

File: index.ts [hunk 3]
```diff
@@ -48,6 +48,30 @@ app.get(
...
+      .then(() =>
```
Comment (nikonikoniko) [inline index.ts:61]: so this `then` gets called before it is done

File: src/store/index.ts (hunk 1 above)
Comment (basilesimon) [reply src/store/index.ts]: That was bang on!

**Pull request merged**